pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -170,8 +170,12 @@ func syncOpenShiftControllerManager_v311_00_to_latest(
 	if actualRCDeployment.Status.AvailableReplicas == 0 {
 		progressingMessages = append(progressingMessages, fmt.Sprintf("deployment/route-controller-manager: available replicas is %d, desired available replicas > %d", actualRCDeployment.Status.AvailableReplicas, 1))
 	}
-	if actualRCDeployment.Status.UpdatedReplicas != *actualRCDeployment.Spec.Replicas {
-		progressingMessages = append(progressingMessages, fmt.Sprintf("deployment/route-controller-manager: updated replicas is %d, desired replicas is %d", actualRCDeployment.Status.UpdatedReplicas, *actualRCDeployment.Spec.Replicas))
+	if actualRCDeployment.Spec.Replicas == nil || actualRCDeployment.Status.UpdatedReplicas != *actualRCDeployment.Spec.Replicas {
```

Comment:
spec.replicas is defaulted by the kube-apiserver, how was this ever nil unless you had an error applying the deployment?

Reply:
that's the case that I noticed in several tests, it failed applying, usually in the early stages of cluster lifecycle

Reply:
See the linked logs